Dockerfile

```diff
@@ -22,25 +22,34 @@ COPY --from=ibm-semeru-runtimes:open-8-jdk-jammy /opt/java/openjdk /usr/lib/jvm/
 COPY --from=ibm-semeru-runtimes:open-11-jdk-jammy /opt/java/openjdk /usr/lib/jvm/semeru11
 COPY --from=ibm-semeru-runtimes:open-17-jdk-jammy /opt/java/openjdk /usr/lib/jvm/semeru17
 
-COPY --from=graalvm-native-image-jdk11 /opt/graalvm-ce-java11-22* /usr/lib/jvm/graalvm22-jdk11
-COPY --from=graalvm-native-image-jdk17 /opt/graalvm-ce-java17-22* /usr/lib/jvm/graalvm22-jdk17
+COPY --from=graalvm-native-image-jdk11 /opt/graalvm-ce-java11-22* /usr/lib/jvm/graalvm11
+COPY --from=graalvm-native-image-jdk17 /opt/graalvm-ce-java17-22* /usr/lib/jvm/graalvm17
 
 RUN sudo apt-get -y update && sudo apt-get -y install curl
 # See: https://gist.github.com/wavezhang/ba8425f24a968ec9b2a8619d7c2d86a6
 RUN set -eux; \
     sudo mkdir -p /usr/lib/jvm/oracle8; \
     curl -L --fail "https://javadl.oracle.com/webapps/download/AutoDL?BundleId=246284_165374ff4ea84ef0bbd821706e29b123" | sudo tar -xvzf - -C /usr/lib/jvm/oracle8 --strip-components 1
 
+# Remove cruft from JDKs that is not used in the build process.
 RUN sudo rm -rf \
     /usr/lib/jvm/*/man \
     /usr/lib/jvm/*/lib/src.zip \
     /usr/lib/jvm/*/demo \
     /usr/lib/jvm/*/sample \
     /usr/lib/jvm/graalvm*/lib/installer
 
-FROM cimg/base:edge-22.04
+FROM cimg/base:edge-22.04 AS base_builder
 
-COPY --from=builder /usr/lib/jvm /usr/lib/jvm
+COPY --from=builder /usr/lib/jvm/8 /usr/lib/jvm/8
+COPY --from=builder /usr/lib/jvm/11 /usr/lib/jvm/11
+COPY --from=builder /usr/lib/jvm/17 /usr/lib/jvm/17
+
+# Base image with minimunm requirenents to build the project.
+# Based on CircleCI Base Image with Ubuntu 22.04.3 LTS, present in most runners.
+FROM cimg/base:edge-22.04 AS base
+
+COPY --from=base_builder /usr/lib/jvm /usr/lib/jvm
```

Comment:
I noticed some weirdness when using `COPY` with multiple source directories. I think it's because of this:

> Multiple <src> resources may be specified but the paths of files and directories will be interpreted as relative to the source of the context of the build.
> [docs](https://docs.docker.com/engine/reference/builder/#copy)

But I'll change it to `scratch`.